### PR TITLE
Fix `NoneType` object error in migration

### PR DESCRIPTION
**Pull request by lanzhenw**
## What changes are proposed in this pull request?

fix the `NoneType` object (colorscheme.fields) is not iterable bug when migrating dataset.

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: fiftyone/migrations/revisions/v0_23_0.py [hunk 1]
```diff
@@ -44,8 +44,10 @@ def up(db, dataset_name):
             )
 
             # Handle nested fields within 'fields'
-            for field in color_scheme["fields"]:
-                field.setdefault("maskTargetsColors", [])
+            if color_scheme["fields"] is not None:
+                for field in color_scheme["fields"]:
```
Comment (benjaminpkane) [inline fiftyone/migrations/revisions/v0_23_0.py:48]:
```suggestion
            for field in color_scheme.get("fields", None):
```

Comment (lanzhenw) [reply fiftyone/migrations/revisions/v0_23_0.py:48]: This will cause the same TypeError for iterating over `None`

Comment (benjaminpkane) [reply fiftyone/migrations/revisions/v0_23_0.py:48]: My mistake, I meant a list `[]`. I worry `color_scheme["fields"]` could throw a key error

File: fiftyone/migrations/revisions/v0_23_0.py [hunk 2]
```diff
@@ -81,6 +82,7 @@ def down(db, dataset_name):
 
             # Handle nested fields within 'fields'
             for field in color_scheme.get("fields", []):
-                field.pop("maskTargetsColors", None)
+                if field is not None:
+                    field.pop("maskTargetsColors", None)
```
Comment (benjaminpkane) [inline fiftyone/migrations/revisions/v0_23_0.py:86]:
```suggestion
                field.pop("maskTargetsColors", None)
```

**Review by benjaminpkane — APPROVED** [review]
LGTM :t

File: fiftyone/migrations/revisions/v0_23_0.py (hunk 1 above)
Comment (lanzhenw) [reply fiftyone/migrations/revisions/v0_23_0.py]: Yeah, just updated.

**Pull request merged**